File: src/WebApp/src/app/Models/CreateEventDto.ts
import { TicketDto } from "./TicketDto";

export class CreatEventDto
{
    constructor(
        private title : string ,
        private description : string ,
        private location : string,
        private startDateTime : Date,
        private endDateTime : Date,
        private capacity : number,
        private category: string,
        private image? : string | null,
        private imageFile? : File | null,
        private tickets? : TicketDto[] | null,
        private parentId? : number | null,
    ){}
}

File: src/Backend/Backend/Models/Dto/CreateEventDto.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models.Dto
{
    public class CreateEventDto
    {
        [Required]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string Location { get; set; }

        [Required]
        public DateTime StartDateTime { get; set; }

        [Required]
        public DateTime EndDateTime { get; set; }
        [Required]
        public EventCategory Category { get; set; }

        [Required]
        public int Capacity { get; set; }

        public string Image { get; set; }
        public IFormFile imageFile { get; set; }

        public List<TicketDto> Tickets { get; set; }
    }
    public class TicketDto
    {
        [Required]
        public string Name { get; set; }

        [Required]
        public decimal Price { get; set; }
        [Required]
        public DateTime validFrom { get; set; }
        [Required]
        public DateTime validUntil { get; set; }
        [Required]
        public int Quota { get; set; }

        [Required]
        public string Description { get; set; }
    }
}

